openfisca_core/tools/test_runner.py

```diff
@@ -212,14 +213,28 @@ def check_variable(self, variable_name, expected_value, period, entity_index = N
 
         actual_value = self.simulation.calculate(variable_name, period)
 
+        absolute_error_margin = self.test.get('absolute_error_margin')
+        variable_absolute_error_margin = (
+            absolute_error_margin.get(variable_name, absolute_error_margin.get("default"))
+            if isinstance(absolute_error_margin, collections.Mapping)
+            else absolute_error_margin
+            )
+
+        relative_error_margin = self.test.get('relative_error_margin')
+        variable_relative_error_margin = (
+            relative_error_margin.get(variable_name, relative_error_margin.get("default"))
+            if isinstance(relative_error_margin, collections.Mapping)
+            else relative_error_margin
+            )
+
         if entity_index is not None:
             actual_value = actual_value[entity_index]
         return assert_near(
             actual_value,
             expected_value,
-            absolute_error_margin = self.test.get('absolute_error_margin'),
+            absolute_error_margin = variable_absolute_error_margin,
             message = f"{variable_name}@{period}: ",
-            relative_error_margin = self.test.get('relative_error_margin'),
+            relative_error_margin = variable_relative_error_margin,
```

Comment:
How did this work previously, do you have an example?

Reply:
Current tests clearly shows how margins can be used today.